src/app/settings/settings.page.ts

```diff
@@ -0,0 +1,86 @@
+import { Component, OnInit, OnDestroy } from '@angular/core';
+import { ModalController, ToastController } from '@ionic/angular';
+import { UserService, VoiceActor, VoiceActorService, PresentationOrder } from 'wanikani-api-ng';
+import { Observable, Subject } from 'rxjs';
+import { map, takeUntil, take } from 'rxjs/operators';
+import { FormGroup, FormControl } from '@angular/forms';
+
+@Component({
+  selector: 'app-settings',
+  templateUrl: './settings.page.html',
+  styleUrls: ['./settings.page.scss'],
+})
+export class SettingsPage implements OnInit, OnDestroy {
+
+  private destroyed = new Subject<boolean>();
+
+  voiceActors: Observable<VoiceActor[]>
+  //WaniKani accepts batch sizes from 3 to 10 inclusive this array is for using ngFor
+  batchSizes = Array(8).fill(1).map((x,i)=>i+3)
+
+  //these are used for iterating over enums in ngFor
+  keys = Object.keys
+  presentationOrderEnum = PresentationOrder
+
+  preferencesForm = new FormGroup({
+    default_voice_actor_id: new FormControl(""),
+    lessons_autoplay_audio: new FormControl(""),
+    lessons_batch_size: new FormControl(""),
+    lessons_presentation_order: new FormControl(""),
+    reviews_autoplay_audio: new FormControl(""),
+    reviews_display_srs_indicator: new FormControl(""),
+  })
+
+  constructor(  private modalController: ModalController, 
+                private userService: UserService, 
+                private voiceActorService: VoiceActorService,
+                private toastController: ToastController) { }
+
+  ngOnDestroy(): void {
```

Comment:
I'm being nit picky, but put ngOnDestroy after ngOnInit

Reply:
this is what I get for letting the IDE generate the methods